src/models/position.ts

```diff
@@ -0,0 +1,44 @@
+import type { UserDoc } from './user';
+
+import { Types, Document, Schema, model } from 'mongoose';
+import { ModelRefs } from './types';
+
+export interface PositionDoc extends Document {
+  title: string;
+  description: string;
+  isOpen: boolean;
+  occupant: Types.ObjectId | UserDoc;
+}
+
+const PositionSchema = new Schema({
+  title: {
+    type: String,
+    required: true,
+    trim: true,
+  },
+  description: {
+    type: String,
+    required: true,
+    trim: true,
+  },
+  isOpen: {
+    type: Boolean,
+    required: true,
+    default: false,
+  },
+  occupant: {
+    type: Types.ObjectId,
+    required: true,
+    ref: ModelRefs.USER,
+  },
+});
+
+PositionSchema.virtual('nomination', {
+  ref: ModelRefs.NOMINATION,
+  localField: '_id',
+  foreignField: 'position',
+});
```

Comment:
Cool so this will give us an array of related nominations on a `PositionDoc` without us having to add them to the Model 10/10

Reply:
should we call this `nominations` plural?

Reply:
Yea this is a bit confusing and I always mix up what this means. (I'll add a comment)
Essentially it means that a `Nomination` has a property called `position` that references this `Position` model.

Reply:
Hmm isn't what you described there the position ref on the Nominations schema? We can have a `Nomination` have a property called `position` that references the `Position` model without this code.

Wouldn't this code actually add a virtual `nomination` array property to the `Position` docs that contains many references to Nominations?

Reply:
Honestly I should do some reading into what this code does because I'm not too sure. I thought it was necessary in defining any relationship😂
But now I'm second guessing

Reply:
Bro I think you're right. we 100% don't need this code🤣

Reply:
🤣  I think it could be useful to have the virtual properties! but yeah not strictly necessary